src/Engine/Graphics/Outdoor.cpp

```diff
@@ -2308,289 +2161,328 @@ void ODM_ProcessPartyActions() {
-                v2 += fixpoint_mul(v118, pODMFace->pFacePlaneOLD.vNormal.x);
-                v128 += fixpoint_mul(v118, pODMFace->pFacePlaneOLD.vNormal.y);
-                fall_speed += fixpoint_mul(v118, pODMFace->pFacePlaneOLD.vNormal.z);
-                if (v2 * v2 + v128 * v128 >= 400) {
+                int dot = abs(party_y_speed * pODMFace->pFacePlaneOLD.vNormal.y +
+                           party_z_speed * pODMFace->pFacePlaneOLD.vNormal.z +
+                           party_x_speed * pODMFace->pFacePlaneOLD.vNormal.x) >> 16;
+                if ((collision_state.speed / 8) > dot)
+                    dot = collision_state.speed / 8;
+                party_x_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.x);
+                party_y_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.y);
+                party_z_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.z);
+                bpartyslope = true;
+                if (party_x_speed * party_x_speed + party_y_speed * party_y_speed >= 400) {
                     if (pParty->floor_face_pid != collision_state.pid &&
                         pODMFace->Pressure_Plate()) {
                         pParty->floor_face_pid = collision_state.pid;
                         trigger_id = pODMFace->sCogTriggeredID;  //
                     }
                 } else {
-                    v2 = 0;
-                    fall_speed = 0;
-                    *(float *)&v128 = 0.0;
+                    party_x_speed = 0;
+                    party_z_speed = 0;
+                    party_y_speed = 0;
                 }
             }
         }
-        v2 = fixpoint_mul(58500, v2);
-        v128 = fixpoint_mul(58500, v128);
-        v122 = fixpoint_mul(58500, v122);
-        fall_speed = fixpoint_mul(58500, fall_speed);
+
+        // ~0.9x reduce party speed and try again
+        party_x_speed = fixpoint_mul(58500, party_x_speed);
+        party_y_speed = fixpoint_mul(58500, party_y_speed);
+        party_z_speed = fixpoint_mul(58500, party_z_speed);
+    }
+    if (bpartyslope)
+        pParty->uFallStartZ = party_new_Z;
+
+    // walking / running sounds ------------------------
+    if (engine->config->settings.WalkSound.Get() && pParty->walk_sound_timer) {
+        if (pParty->walk_sound_timer >= pEventTimer->uTimeElapsed)
+            pParty->walk_sound_timer -= pEventTimer->uTimeElapsed;
+        else
+            pParty->walk_sound_timer = 0;
     }
 
-    // Воспроизведение звуков ходьбы/бега------------------------
-    uint pX_ = abs(pParty->vPosition.x - pX);
-    uint pY_ = abs(pParty->vPosition.y - pY);
-    uint pZ_ = abs(pParty->vPosition.z - party_new_Z);
+    // save up distance deltas so walks sounds play at high fps with small delta
+    int pX_ = pParty->vPosition.x - party_new_x;
+    int pY_ = pParty->vPosition.y - party_new_Y;
+    int pZ_ = pParty->vPosition.z - party_new_Z;
+    pParty->_movementTallySQ += pX_ * pX_ + pY_ * pY_ + pZ_ * pZ_;
```

Comment:
Sum of squares != square of sum. So you'll need sqrt call here.

Reply:
Ooof.